### PR TITLE
Lf 2525 handle asynchronous add sensor outcome

### PR DESCRIPTION
**Pull request by Cali0707**
This PR handles the sync/async outcomes based off how long the flow is taking.

To test:
1. In `sensorController.js` on line 42, set the timeout to something small, like `50`
2. Comment out line 134 in `sensorController.js` as the webhook won't register properly until we are on beta/avocado
3. Try uploading a test csv file - you should see an error snackbar appear, this is because the 202 response code is not currently handled on the frontend in this branch - that is being handled in #2099 
4. Go to the notification that has been generated, it should have a "take me there" button which either a) downloads a text file if there were errors, or b) takes you to the map if it was successful

A useful testing file is: 
[testBulkUpload.csv](https://github.com/LiteFarmOrg/LiteFarm/files/9002688/testBulkUpload.csv)

### CODE REVIEW

**Review by fsayyid — APPROVED** [review]
From a backend perspective, this seems to be working well 👍

File: packages/api/src/controllers/sensorController.js [hunk 1]
```diff
@@ -139,17 +179,49 @@ const sensorController = {
...
+            : res
+                .status(200)
+                .send({ message: 'Successfully uploaded!', sensors: sensorLocations }) &&
+                clearTimeout(timer);
         }
       }
     } catch (e) {
       console.log(e);
-      res.status(500).send(e.message);
+      const { user_id } = req.user;
```
Comment (smattingly) [inline packages/api/src/controllers/sensorController.js:216]: `user_id` and `farm_id` are already defined, just not in this scope. Move the definitions out of the `try` block and eliminate the repetition.

File: packages/api/src/controllers/sensorController.js [hunk 2]
```diff
@@ -30,6 +30,14 @@ const sensorErrors = require('../util/sensorErrors');
 
 const sensorController = {
   async addSensors(req, res) {
+    let hasTimedOut = false;
```
Comment (smattingly) [inline packages/api/src/controllers/sensorController.js:33]: Technically this comment is for the line above the one marked:
See [simplify-routing](https://lite-farm.atlassian.net/wiki/spaces/LITEFARM/pages/1191575576/Pre-integration+peer+review#Backend-code): avoid functions that return constant functions.

**Review by smattingly — CHANGES_REQUESTED** [review]
Requesting a couple of improvements.

File: packages/api/src/controllers/sensorController.js (hunk 1 above)
Comment (Cali0707) [reply packages/api/src/controllers/sensorController.js:216]: Resolved

File: packages/api/src/controllers/sensorController.js (hunk 2 above)
Comment (Cali0707) [reply packages/api/src/controllers/sensorController.js:33]: @smattingly I'm not sure what you are referring to here - as far as I can tell `addSensors` is not returning a constant function?

Comment (smattingly) [reply packages/api/src/controllers/sensorController.js:33]: Sorry, you're right. I guess I just entirely misread it.

**Review by smattingly — APPROVED** [review]

**Pull request merged**